library/systemd/src/lib/yast2/compound_service.rb

```diff
@@ -0,0 +1,202 @@
+    #     for services that does not have socket, it starts on boot.
+    #   - `:manual` all services and all its associated sockets is disabled.
+    #   - `:inconsistent` mixture of states
+    # @see Yast::SystemService#current_start_mode
+    # @note additional state `:inconsistent` that is not in SystemService
+    def current_start_mode
+      @current_start_mode ||= services_mode(method(:has_current_mode))
+    end
+
+    # returns configuration for start mode.
+    # @return [:on_boot, :on_demand, :manual, :inconsistent] start_mode configuration:
+    #
+    #   - `:on_boot` when start mode is set to on_boot for all services.
+    #   - `:on_demand` when start mode is set to on_demand for services that support it and on_boot
+    #     otherwise.
+    #   - `:manual` when start mode is set to manual for all services.
+    #   - `:inconsistent` when services have mixture of start modes
+    # @see Yast::SystemService#start_mode
+    # @note additional state `:inconsistent` that is not in SystemService
+    def start_mode
+      services_mode(method(:has_mode))
+    end
+
+    AUTOSTART_OPTIONS = [:on_boot, :on_demand, :manual, :inconsistent].freeze
+    # sets start mode configuration.
+    # @param [:on_boot, :on_demand, :manual, :inconsistent] sets start_mode.
+    #   See {Yast2::SystemService.start_mode=}. Additionally inconsistent is used
+    #   to keep current start mode. For each service.
+    def start_mode=(configuration)
+      if !AUTOSTART_OPTIONS.include?(configuration)
+        raise ArgumentError, "Invalid parameter #{configuration.inspect}"
+      end
+
+      if configuration == :inconsistent
+        reset(exclude: [:action])
+      else
+        services.all { |s| s.start_mode = configuration }
+      end
+    end
+
+    # returns true if any allow start on demand
+    def support_start_on_demand?
+      @services.any?(&:support_start_on_demand?)
+    end
+
+    # resets changes.
+    # @param exclude [Array] exclude from reset some parts. Now supported: `:action` and
+    #   `:start_mode`.
+    def reset(exclude: [])
+      old_action = action
+      old_start_mode = start_mode
+      services.all(&:reset)
+      @current_start_mode = nil
+      public_send(old_action) if old_action != nil && !exclude.include?(:action)
+      if old_start_mode != :inconsistent && !exclude.include?(:start_mode)
+        self.start_mode = old_start_mode
+      end
+    end
+
+    def start
+      services.each(&:start)
+    end
+
+    def stop
+      services.each(&:stop)
+    end
+
+    def restart
+      services.each(&:restart)
+    end
+
+    def reload
+      services.each(&:reload)
+    end
+
+  private
+
+    def has_current_mode(start_mode)
+      ->(service) { service.current_start_mode == start_mode }
+    end
+
+    def has_mode(start_mode)
+      ->(service) { service.start_mode == start_mode }
+    end
+
+    def services_mode(mode_method)
+      if services.all?(&mode_method.call(:enabled?))
```

Comment:
`:enabled?` ?

Reply:
good catch, I use wrong symbol. It should be `:on_boot`